File: 練習/3/Sphere.cpp
#include "Sphere.h"
#include <sstream>
#include <cmath>

Sphere::Sphere() {

}
Sphere::Sphere(double) {

}
Sphere::Sphere(double r, string c)
{
	this->radius = r;
	setColor(c);
}

void Sphere::setRadius(double r)
{
	this->radius = r;
}

double Sphere::getRadius()
{
	return radius;
}

double Sphere::getVolume()
{
	return (3.14159 * pow(radius, 3)) * 4 / 3;
}

string Sphere::toString()
{
	return "Sphere:";
}

File: 練習/6/Sphere.py
from Shape import Shape

class Sphere(Shape):
    def __init__(self, radius, color):
        super().__init__(color)
        self.__radius = radius
        
    def setRadius(self, radius):
        self.__radius = radius
        
    def getRadius(self):
        return self.__radius
    
    def getVolume(self):
        volume = 4 * 3.14159 * pow(self.getRadius(), 3) / 3
        return round(volume, 1)
    
    def __str__(self):
        return "Sphere:"